File: 2021/src/test/java/de/jfriemel/aoc/days/Day02Test.java
package de.jfriemel.aoc.days;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Day02Test {

    private static final String INPUT_STRING = """
            forward 5
            down 5
            forward 8
            up 3
            down 8
            forward 2""";

    private static final List<String> INPUT = Arrays.asList(INPUT_STRING.split("\n"));

    @Test
    public void testPart1() {
        String result = new Day02().part1(INPUT);
        assertEquals("150", result);
    }

    @Test
    public void testPart2() {
        String result = new Day02().part2(INPUT);
        assertEquals("900", result);
    }

}

File: 2023/src/test/kotlin/de/jfriemel/aoc/days/Day02Test.kt
package de.jfriemel.aoc.days

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Day02Test {
    private val input = """
        Game 1: 3 blue, 4 red; 1 red, 2 green, 6 blue; 2 green
        Game 2: 1 blue, 2 green; 3 green, 4 blue, 1 red; 1 green, 1 blue
        Game 3: 8 green, 6 blue, 20 red; 5 blue, 4 red, 13 green; 5 green, 1 red
        Game 4: 1 green, 3 red, 6 blue; 3 green, 6 red; 3 green, 15 blue, 14 red
        Game 5: 6 red, 1 blue, 3 green; 2 blue, 1 red, 2 green
    """.trimIndent().lines()

    @Test
    fun testPart1() {
        val result = Day02.part1(input)
        Assertions.assertEquals("8", result)
    }

    @Test
    fun testPart2() {
        val result = Day02.part2(input)
        Assertions.assertEquals("2286", result)
    }
}
